rpc/gnmi/gnmi-specification.md

```diff
@@ -688,20 +690,101 @@ the catalog entry.
+input.
+
+CLI configuration is defined as the command line interface text used for
+configuration.  In gNMI, this is encoded as ASCII text.  The format of the ASCII
+text is agreed to out of band from  gNMI.  The string used to identify the CLI
+origin must be formatted as `origin:<nos_cli>`.  Example values might include:
+`xros_cli`, `junos_cli`, `srlinux_cli` or any string which indicates the NOS
+name and the CLI designation.
+
+### 2.7.2 Definition of `origin` for YANG-modelled Data
+
+The `openconfig-extensions:origin` field MAY be utilised to determine the
+origin within which a particular module is instantiated. The specification of
+this extension is within
+[openconfig-extensions.yang](https://github.com/openconfig/public/blob/master/release/models/openconfig-extensions.yang).
+
+It should be noted that `origin` is distinct from `namespace`. Whilst a YANG
+namespace is defined at any depth within the schema tree, an `origin` is
+only used to disambiguate entire schema trees. That is to say, any element
+that is not at the root inherits its `origin` from its root entity, regardless
+of the YANG schema modules that make up that root.
+
+### 2.7.3 Partial Specifications of Origin in Set
+
+If a `Set` RPC specifies `delete`, `update`, or `replace` fields which include
+an `origin` within their `Path` messages, the corresponding change MUST be
+constrained to the specified origin. Particularly:
+
+* `replace` operations MUST only replace the contents of the specified `origin`
+  at the specified path. Origins that are not specified within the `SetRequest`
+  MUST NOT have their contents replaced. In order for a `replace` operation to
+  replace any contents of an `origin` it must be explicitly specified in the
+  `SetRequest`.
+* `delete` operations MUST delete only the contents at the specified path within
+  the specified `origin`. To delete contents from multiple origins, a client
+  MUST specify multiple paths within the `delete` of the `SetRequest`.
+
+### 2.7.4 Transactionality of Sets with multiple Origins
+
+Where a `SetRequest` specifies more than one `origin` - i.e., two or more
+operations whose path include more than one origin - manipulations to all
+affected trees MUST be considered as a single transaction. That is to say, only
+if all transactions succeed should the `SetResponse` indicate success. If any
+of the transactions fail, the contents of all origins MUST be rolled back, and
+an error status returned upon responding to the `Set` RPC.
+
+## 2.8 Extensions to gNMI
 
 Each top-level RPC message (e.g., `SubscribeRequest` and `SubscribeResponse` for
 the `Subscribe` RPC) defines an `extensions` field which can be used to carry
 additional parameters for a gNMI RPC.
-[[GNMI-EXT]](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-extensions.md)
+[[GNMI-EXT]](<https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-extensions.md>)
```

Comment:
This doesn't look correct.

Reply:
Apparently markdown lint md034 - bare url says there should be <> enclosing a URL.  I haven't noticed that before myself.  The link is working when rendered in github and VSCode.  So I guess I'll leave it?

Reply:
I think the linter I am using is not so good on this point and the rationale for this not so strong, so I have removed the angle brackets for URL's for consistency across the doc.